pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -604,8 +603,11 @@ func (oe *operationExecutor) IsOperationPending(volumeName v1.UniqueVolumeName,
 func (oe *operationExecutor) AttachVolume(
 	volumeToAttach VolumeToAttach,
 	actualStateOfWorld ActualStateOfWorldAttacherUpdater) error {
-	generatedOperations :=
+	generatedOperations, err :=
```

Comment:
I think we want to avoid pushing errors at this level since these errors wont be retried with exponential backoff. Is there any way to push these errors into the generated function instead?

You can see in some of the existing `Generate...VolumeFunc`'s that we ignore errors and just log something instead (this is not ideal but a last resort for when we couldn't push the error into the generated func).

Reply:
Most of the other generator functions invoked by operationExecutor like DetachVolume/[GenerateDetachVolumeFunc](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_executor.go#L618), MountVolume/[GenerateMapVolumeFunc](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_executor.go#L771), UnmountVolume/[GenerateUnmountVolumeFunc](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_executor.go#L813), etc. seem to return error from the generators that are bubbled up. The change here aligns AttachVolume/GenerateAttachVolumeFunc (as well as MountVolume/GenerateMountVolumeFunc) with that convention.

Reply:
IIUC we were actually in a transition period trying to go in the other direction: https://github.com/kubernetes/kubernetes/issues/74988

Reply:
Thanks for the context. Looks like I was undoing https://github.com/kubernetes/kubernetes/pull/75986/ . Will ammend.